client/state/stats/posts/reducer.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import get from 'lodash/get';
+
+/**
+ * Internal dependencies
+ */
+import { isValidStateWithSchema } from 'state/utils';
+import { items as itemSchemas } from './schema';
+import {
+	SERIALIZE,
+	DESERIALIZE,
+	POST_STATS_RECEIVE,
+	POST_STATS_REQUEST,
+	POST_STATS_REQUEST_FAILURE,
+	POST_STATS_REQUEST_SUCCESS
+} from 'state/action-types';
+
+/**
+ * Returns the updated requests state after an action has been dispatched. The
+ * state maps site ID, post ID and stat keys to whether a request is in progress.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function requesting( state = {}, action ) {
+	switch ( action.type ) {
+		case POST_STATS_REQUEST:
+		case POST_STATS_REQUEST_SUCCESS:
+		case POST_STATS_REQUEST_FAILURE:
+			return Object.assign( {}, state, {
```

Comment:
Would [`merge`](http://lodash.com/docs#merge) have helped here to avoid the layers of `Object.assign` ?

```js
return merge( {}, state, {
	[ action.siteId ]: {
		[ action.postId ]: {
			[ action.stat ]: POST_STATS_REQUEST === action.type
		}
	}
} );
```

Also, `Object.assign` is happy to accept `undefined` so long as it's not the first argument, so you could omit the `{}` fallback value to `get`. Example:

```js
console.log( Object.assign( { a: 1 }, undefined, undefined, { b: 2 }, undefined ) );
// -> { a: 1, b: 2 }
```